### ISSUE TITLE
[Feature]: Support Ubuntu 24.04

### ISSUE THREAD

**Issue by Mister-Hope** [microsoft/playwright#30368]
### 🚀 Feature Request

Ubuntu 24 will be released in this month, and the beta version is already release in 12th. (See https://wiki.ubuntu.com/Releases)

Hope playweight can add support once it release

### Motivation

Cuurently when running `pnpm exec playwright install --with-deps` on Ubuntu 24.04, errors are thrown

```
Hit:1 http://security.ubuntu.com/ubuntu noble-security InRelease
Get:2 http://archive.ubuntu.com/ubuntu noble InRelease [255 kB]
Get:3 http://archive.ubuntu.com/ubuntu noble-updates InRelease [89.7 kB]
Hit:4 http://archive.ubuntu.com/ubuntu noble-backports InRelease
Get:5 http://archive.ubuntu.com/ubuntu noble/main amd64 Packages [1404 kB]
Get:6 http://archive.ubuntu.com/ubuntu noble/universe amd64 Packages [15.2 MB]
Get:7 http://archive.ubuntu.com/ubuntu noble/universe Translation-en [6029 kB]
Get:8 http://archive.ubuntu.com/ubuntu noble/multiverse amd64 Packages [241 kB]                                                                                          
Fetched 23.2 MB in 7s (3262 kB/s)                                                                                                                                        
Reading package lists... Done
Reading package lists... Done
Building dependency tree... Done
Reading state information... Done
Note, selecting 'libatk-bridge2.0-0t64' instead of 'libatk-bridge2.0-0'
Note, selecting 'libatk1.0-0t64' instead of 'libatk1.0-0'
Note, selecting 'libatspi2.0-0t64' instead of 'libatspi2.0-0'
Note, selecting 'libcups2t64' instead of 'libcups2'
Note, selecting 'libgtk-3-0t64' instead of 'libgtk-3-0'
Note, selecting 'libpng16-16t64' instead of 'libpng16-16'
Note, selecting 'libevent-2.1-7t64' instead of 'libevent-2.1-7'
Package libasound2 is a virtual package provided by:
  liboss4-salsa-asound2 4.2-build2020-1ubuntu3
  libasound2t64 1.2.11-1build2 (= 1.2.11-1build2)
You should explicitly select one to install.

E: Package 'libasound2' has no installation candidate
E: Unable to locate package libicu70
E: Unable to locate package libffi7
E: Unable to locate package libx264-163
Failed to install browsers
Error: Installation process exited with code: 100
```

**Comment by mxschmitt** [comment]
We are currently blocked by GitHub Actions, that they add support for it in their [runner-images](https://github.com/actions/runner-images) repository.

**Comment by man-qa** [comment]
Same for me, stuck on ubuntu:22.04 for the moment

**Comment by ajrowland** [comment]
I can run on 24.04 via the UI, or just using the `--project firefox` flag.

**Comment by rotemgrim** [comment]
I'm stuck with ubuntu 24.04. any one has a workaround maybe?

**Comment by mwanago** [comment]
> I'm stuck with ubuntu 24.04. any one has a workaround maybe?

I was able to run it on 23.10 with this advice:
https://github.com/microsoft/playwright/issues/23296#issuecomment-1670807705

I had to compile `libx264-163` from the source code though.

**Comment by wangfuxin8080** [comment]
hi @mxschmitt , do you know when the github issue will be fixed, I encouter the same error when running 
apt-get update && apt-get upgrade -y && apt-get install -y libxtst6 libnss3 libcups2 libxss1 libxrandr2 libasound2 libatk1.0-0 libgtk-3-0

error:
Hit:1 http://archive.ubuntu.com/ubuntu noble InRelease
Hit:2 http://archive.ubuntu.com/ubuntu noble-updates InRelease
Hit:3 http://archive.ubuntu.com/ubuntu noble-backports InRelease
Hit:4 http://security.ubuntu.com/ubuntu noble-security InRelease
Reading package lists...
Reading package lists...
Building dependency tree...
Reading state information...
Calculating upgrade...
0 upgraded, 0 newly installed, 0 to remove and 0 not upgraded.
Reading package lists...
Building dependency tree...
Reading state information...
Package libasound2 is a virtual package provided by:
  liboss4-salsa-asound2 4.2-build2020-1ubuntu3
  libasound2t64 1.2.11-1build2 (= 1.2.11-1build2)

E: Package 'libasound2' has no installation candidate

**Comment by LostInDarkMath** [comment]
I have the same issue. Any news on this?

**Comment by WithMarcel** [comment]
I encountered a similar issue on Ubuntu 24.04, albeit with a wider range of missing packages: `libicu70`, `libffi7`, and `libx264-163`. To resolve this, I downloaded them from https://packages.ubuntu.com/, specifically for the 22.04 version. While this method isn't advisable due to its deviation from best practices, I managed to install them using `apt install ./<package>.deb`.

For `libasound2`, I initially attempted to install `libasound2t64`, only to discover it was already present. Subsequently, I checked `liboss4-salsa-asound2`, but refrained from installing it as it would entail the removal of over 70 other packages, which was not my intended outcome.

Proceeding with `playwright install --with-deps`, I encountered additional absent packages: `libvpx7` and `libevent-2.1-7`. After obtaining and installing them, I was finally able to execute `playwright install` without encountering any further issues.

It's worth noting that while this workaround worked for me, I don't recommend it universally, nor can I guarantee its seamless functionality for all users. Probably as soon as Ubuntu 24.04 is officially supported, I would try to remove the above packages and reinstall playwright.

**Comment by Mister-Hope** [comment]
> We are currently blocked by GitHub Actions, that they add support for it in their [runner-images](https://github.com/actions/runner-images) repository: [actions/runner-images#9691](https://github.com/actions/runner-images/issues/9691).


It seems that there is still a time period before GitHub support 24.04.

It would be great that we can have Ubuntu 24.04 supported in playwright first, then adding a CI test once it supports.

Even a workaround is better than no support.

**Comment by jamieredding** [comment]
Hi @WithMarcel, I've made progress with the steps you provided so thank you very much.

I encountered exactly your issue when reaching this point:
> For `libasound2`, I initially attempted to install `libasound2t64`, only to discover it was already present. Subsequently, I checked `liboss4-salsa-asound2`, but refrained from installing it as it would entail the removal of over 70 other packages, which was not my intended outcome.

However `playwright install --with-deps` still continues to complain about `libasound2 ` not being installed, instead of encountering the next round of absent packages that you ran into here: 
> Proceeding with `playwright install --with-deps`, I encountered additional absent packages: `libvpx7` and `libevent-2.1-7`. After obtaining and installing them, I was finally able to execute `playwright install` without encountering any further issues.

Is there anything you did between these steps that helped you progress?

**Comment by MarcosIgnacioo** [comment]
wep ya lo arreglaron??

**Comment by NoahNim** [comment]
> Hi @WithMarcel, I've made progress with the steps you provided so thank you very much.
> 
> I encountered exactly your issue when reaching this point:
> 
> > For `libasound2`, I initially attempted to install `libasound2t64`, only to discover it was already present. Subsequently, I checked `liboss4-salsa-asound2`, but refrained from installing it as it would entail the removal of over 70 other packages, which was not my intended outcome.
> 
> However `playwright install --with-deps` still continues to complain about `libasound2 ` not being installed, instead of encountering the next round of absent packages that you ran into here:
> 
> > Proceeding with `playwright install --with-deps`, I encountered additional absent packages: `libvpx7` and `libevent-2.1-7`. After obtaining and installing them, I was finally able to execute `playwright install` without encountering any further issues.
> 
> ### Further steps
> 
> I proceeded to instead run `npx playwright install` without `--with-deps` and I reached the `libvpx7` and `libevent-2.1.7` missing packages that are mentioned above.
> 
> I then installed them manually the same way as the earlier ones by visiting https://packages.ubuntu.com/.
> 
> After that point, `npx playwright install` ran without errors.
> ### Running tests
> 
> I am able to run `npx playwright test` on the example tests generated when you add playwright to a project. 5/6 pass except `[webkit] › example.spec.ts:10:5 › get started link` fails when trying to click a link.
> 
> I'll proceed for now despite this as I'm at least able to test with chromium.

many of the missing dependencies seem to not exist on that site. I am missing the following dependencies according to errors: 
libicudata.so.70 
libicui18n.so.70 
libicuuc.so.70         
libvpx.so.7        
libflite.so.1                                 
libflite_usenglish.so.1               
libflite_cmu_grapheme_lang.so.1                 
libflite_cmu_grapheme_lex.so.1              
libflite_cmu_indic_lang.so.1                
libflite_cmu_indic_lex.so.1                 
libflite_cmulex.so.1                          
libflite_cmu_time_awb.so.1                
libflite_cmu_us_awb.so.1                       
libflite_cmu_us_kal16.so.1            
libflite_cmu_us_kal.so.1                    
libflite_cmu_us_rms.so.1                   
libflite_cmu_us_slt.so.1

and only a few exist on https://packages.ubuntu.com/

**Comment by NoahNim** [comment]
Hello, it seems that the --install-deps command and playwright has issues getting the packages from what's in the default ubuntu.sources file in /etc/apt/sources.list.d

I updated my ubuntu.sources file in /etc/apt/sources.list.d to look like this:

`Types: deb
URIs: http://us.archive.ubuntu.com/ubuntu/
Suites: noble noble-updates noble-backports
Components: main restricted universe multiverse
Signed-By: /usr/share/keyrings/ubuntu-archive-keyring.gpg

Types: deb
URIs: http://security.ubuntu.com/ubuntu/
Suites: noble-security
Components: main restricted universe multiverse
Signed-By: /usr/share/keyrings/ubuntu-archive-keyring.gpg

Types: deb
URIs: http://mirrors.kernel.org/ubuntu/
Suites: focal
Components: main universe
Signed-By: /usr/share/keyrings/ubuntu-archive-keyring.gpg`

This got every dependency to install except libicu70 and libx264-163. Ubuntu 24.04 uses libicu74 and libx264-164, so libicu70 and libx264-163 can't be installed unless a user reverts to those versions, but I don't want to break other things by doing this.

**Comment by vkWeb** [comment]
@mxschmitt please try getting playwright ready for Ubuntu 24.04. Thanks!

**Comment by mxschmitt** [comment]
This will be available in v1.45. Experimental support has been landed on our Canary channel: https://playwright.dev/docs/canary-releases

**Comment by FullyRemoteDev** [comment]
> This will be available in v1.45. Experimental support has been landed on our Canary channel: https://playwright.dev/docs/canary-releases

is a canary release available for Python?

**Comment by jhenriquez-ax** [comment]
1) Updrading playwright to `1.44.1` 

2) Adding these repositories: 

deb http://cz.archive.ubuntu.com/ubuntu jammy main universe
deb http://cz.archive.ubuntu.com/ubuntu jammy main main

3) Installing the missing dependencies:

apt-get update 
sudo apt-get install libicu70 libffi7 libx264-163 libvpx7 libevent-2.1

4) run `playwright install`

This allowed me to run my suites without issues. But running `with-deps` still fails. Hope this helps.

**Comment by Martin-Thiele** [comment]
Also looking for a python solution

**Comment by mirao** [comment]
Has anybody tried WebKit with Playwright 1.45 (where support for Ubuntu 24.04) is?
I'm getting this error on Ubuntu 24.04:
```
$ pnpm update playwright
$ npx playwright --version
Version 1.45.0
$ pnpm playwright install --with-deps
... no error
$  pnpm playwright wk
Error: t.parse: Target page, context or browser has been closed
Browser logs:

<launching> /home/mirao/.cache/ms-playwright/webkit-2035/pw_run.sh --inspector-pipe --no-startup-window
<launched> pid=242596
[pid=242596][err] /home/mirao/.cache/ms-playwright/webkit-2035/minibrowser-gtk/bin/MiniBrowser: error while loading shared libraries: libicudata.so.70: cannot open shared object file: No such file or directory
[pid=242596] <process did exit: exitCode=127, signal=null>
[pid=242596] starting temporary directories cleanup
Call log:
  - <launching> /home/mirao/.cache/ms-playwright/webkit-2035/pw_run.sh --inspector-pipe --no-startup-window
  - <launched> pid=242596
  - [pid=242596][err] /home/mirao/.cache/ms-playwright/webkit-2035/minibrowser-gtk/bin/MiniBrowser: error while loading shared libraries: libicudata.so.70: cannot open shared object file: No such file or directory
  - [pid=242596] <process did exit: exitCode=127, signal=null>
  - [pid=242596] starting temporary directories cleanup
  ```
Chromium and Firefox work fine.

**Comment by mirao** [comment]
> I'm getting this error on Ubuntu 24.04:

Well, it's an issue only on one of my machines with Ubuntu 24.04. Today I upgraded 22.04 => 24.04 on that machine and hit the issue with `libicudata.so.70` there.

Then I tested it on two more machines where I upgraded Ubuntu 23.10 => 24.10 and everything works fine. Also it works well on a VM with clear installation of Ubuntu 24.04 (a multipass VM).

I'll check again what could be wrong on that specific station after upgrade 22.04 => 24.04.

**Comment by mirao** [comment]
> [pid=242596][err] /home/mirao/.cache/ms-playwright/webkit-2035/minibrowser-gtk/bin/MiniBrowser: error while loading shared libraries: libicudata.so.70: cannot open shared object file: No such file or directory
...
> I'll check again what could be wrong on that specific station after upgrade 22.04 => 24.04.

Finally solved by removing of `~/.cache/ms-playwright`, `node_modules` and PNPM store. Then installed Playwright and browsers again and now everything works fine.

**Comment by FullyRemoteDev** [comment]
So finally we have the 1.45 release. thanks.

Now it has to be updated in pip I guess as it still shows 1.44 version. (https://pypi.org/project/playwright/)

or is there anyway I can get this latest 1.45 version in python apart from pip?

**Comment by Martin-Thiele** [comment]
```bash
pip install git+https://github.com/microsoft/playwright-python.git
playwright install-deps
playwright install
```

this worked for me, as @fawazahmed0 suggested. It installs version: `1.44.1.dev11+gc6cc4c9`

**Comment by islam0000** [comment]
Still I'm facing same issue with playwright, java 
<img width="483" alt="image" src="https://github.com/user-attachments/assets/b6ddc064-53db-4a13-94d1-e8dc4c1ab656">

**Comment by silverwind** [comment]
> Still I'm facing same issue with playwright, java <img alt="image" width="483" src="https://private-user-images.githubusercontent.com/18109291/355078371-b6ddc064-53db-4a13-94d1-e8dc4c1ab656.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjMyMDAwMTYsIm5iZiI6MTcyMzE5OTcxNiwicGF0aCI6Ii8xODEwOTI5MS8zNTUwNzgzNzEtYjZkZGMwNjQtNTNkYi00YTEzLTk0ZDEtZThkYzRjMWFiNjU2LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA4MDklMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwODA5VDEwMzUxNlomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWNhOTcwMzc1ZmE3MjY3MmIzOGI5NmMyOTk5ZjJlZTBjNjNhNzc4NTE3ZWY4MTRkNTI3YmRhODU3MGU1ZmFjNzUmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.bSDp25_T8GODUiBehQqTg_j1jFDwl0QWBBuRD5Twf3k">

Should probably open a new issue for this. I also observed it and downgraded to Ubuntu 22 for now.

**Comment by islam0000** [comment]
> > Still I'm facing same issue with playwright, java <img alt="image" width="483" src="https://private-user-images.githubusercontent.com/18109291/355078371-b6ddc064-53db-4a13-94d1-e8dc4c1ab656.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjMyMDAwMTYsIm5iZiI6MTcyMzE5OTcxNiwicGF0aCI6Ii8xODEwOTI5MS8zNTUwNzgzNzEtYjZkZGMwNjQtNTNkYi00YTEzLTk0ZDEtZThkYzRjMWFiNjU2LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA4MDklMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwODA5VDEwMzUxNlomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWNhOTcwMzc1ZmE3MjY3MmIzOGI5NmMyOTk5ZjJlZTBjNjNhNzc4NTE3ZWY4MTRkNTI3YmRhODU3MGU1ZmFjNzUmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.bSDp25_T8GODUiBehQqTg_j1jFDwl0QWBBuRD5Twf3k">
> 
> Should probably open a new issue for this. I also observed it and downgraded to Ubuntu 22 for now.

I have not configured Ubuntu anywhere, shall I change the playwright or chrome version?

**Comment by christopher-r-doucet** [comment]
Just started getting this issue this morning as part of my CICD pipeline, was working completely fine just hours ago: 

```
Fetched 2417 kB in 1s (3691 kB/s)
Reading package lists...
Reading package lists...
Building dependency tree...
Reading state information...
Package libicu70 is not available, but is referred to by another package.
This may mean that the package is missing, has been obsoleted, or
is only available from another source

Package libasound2 is a virtual package provided by:
  liboss4-salsa-asound2 4.2-build2020-1ubuntu3
  libasound2t64 1.2.11-1build2 (= 1.2.11-1build2)

E: Package 'libasound2' has no installation candidate
E: Package 'libicu70' has no installation candidate
E: Unable to locate package libffi7
E: Unable to locate package libx264-163
Failed to install browsers
Error: Installation process exited with code: 100
Error: Process completed with exit code 1.
```

**Comment by Asaf51** [comment]
> Just started getting this issue this morning as part of my CICD pipeline, was working completely fine just hours ago:
> 
> ```
> Fetched 2417 kB in 1s (3691 kB/s)
> Reading package lists...
> Reading package lists...
> Building dependency tree...
> Reading state information...
> Package libicu70 is not available, but is referred to by another package.
> This may mean that the package is missing, has been obsoleted, or
> is only available from another source
> 
> Package libasound2 is a virtual package provided by:
>   liboss4-salsa-asound2 4.2-build2020-1ubuntu3
>   libasound2t64 1.2.11-1build2 (= 1.2.11-1build2)
> 
> E: Package 'libasound2' has no installation candidate
> E: Package 'libicu70' has no installation candidate
> E: Unable to locate package libffi7
> E: Unable to locate package libx264-163
> Failed to install browsers
> Error: Installation process exited with code: 100
> Error: Process completed with exit code 1.
> ```

Same. Is there anything we can do?

**Comment by corneliusroemer** [comment]
I've made a Stackoverflow Q&A to increase discoverability of the fixes (upgrading Playwright to 1.48 or downgrading Ubuntu to 22): https://stackoverflow.com/q/79090211/7483211

**Comment by christopher-r-doucet** [comment]
> Just started getting this issue this morning as part of my CICD pipeline, was working completely fine just hours ago:
> 
> ```
> Fetched 2417 kB in 1s (3691 kB/s)
> Reading package lists...
> Reading package lists...
> Building dependency tree...
> Reading state information...
> Package libicu70 is not available, but is referred to by another package.
> This may mean that the package is missing, has been obsoleted, or
> is only available from another source
> 
> Package libasound2 is a virtual package provided by:
>   liboss4-salsa-asound2 4.2-build2020-1ubuntu3
>   libasound2t64 1.2.11-1build2 (= 1.2.11-1build2)
> 
> E: Package 'libasound2' has no installation candidate
> E: Package 'libicu70' has no installation candidate
> E: Unable to locate package libffi7
> E: Unable to locate package libx264-163
> Failed to install browsers
> Error: Installation process exited with code: 100
> Error: Process completed with exit code 1.
> ```

Already has been discussed but I fixed it this morning by updating Playwright to the latest version.

**Comment by jankalinic** [comment]
> > Just started getting this issue this morning as part of my CICD pipeline, was working completely fine just hours ago:
> > ```
> > Fetched 2417 kB in 1s (3691 kB/s)
> > Reading package lists...
> > Reading package lists...
> > Building dependency tree...
> > Reading state information...
> > Package libicu70 is not available, but is referred to by another package.
> > This may mean that the package is missing, has been obsoleted, or
> > is only available from another source
> > 
> > Package libasound2 is a virtual package provided by:
> >   liboss4-salsa-asound2 4.2-build2020-1ubuntu3
> >   libasound2t64 1.2.11-1build2 (= 1.2.11-1build2)
> > 
> > E: Package 'libasound2' has no installation candidate
> > E: Package 'libicu70' has no installation candidate
> > E: Unable to locate package libffi7
> > E: Unable to locate package libx264-163
> > Failed to install browsers
> > Error: Installation process exited with code: 100
> > Error: Process completed with exit code 1.
> > ```
> 
> Already has been discussed but I fixed it this morning by updating Playwright to the latest version.

+1 From me. I was using java playwright version 1.47.0 on ubuntu 24-04. So i just switched to playwright 1.47.0 (latest on maven repository) and the issue seems to go away.

**Comment by turowicz** [comment]
`yarn playwright install --with-deps chromium`

errors out with:

```
#17 18.08 The following packages have unmet dependencies:
#17 18.66  libglib2.0-0t64 : Breaks: libglib2.0-0 (< 2.80.0-6ubuntu3~)
#17 18.67 E: Unable to correct problems, you have held broken packages.
#17 18.68 Failed to install browsers
```

**Comment by orestis** [comment]
I ended up on this thread when our browser tests were failing on GitHub Actions. Bumping playwright to 1.49 fixed the installation issues, but it seems now there's a Chromium issue:

```
19 12 2024 15:58:18.103:ERROR [launcher]: Cannot start ChromiumHeadless
	[5859:5859:1219/155817.767911:FATAL:zygote_host_impl_linux.cc(128)] No usable sandbox! If you are running on Ubuntu 23.10+ or another Linux distro that has disabled unprivileged user namespaces with AppArmor, see https://chromium.googlesource.com/chromium/src/+/main/docs/security/apparmor-userns-restrictions.md. Otherwise see https://chromium.googlesource.com/chromium/src/+/main/docs/linux/suid_sandbox_development.md for more information on developing with the (older) SUID sandbox. If you want to live dangerously and need an immediate workaround, you can try using --no-sandbox.
[1219/155817.775548:ERROR:file_io_posix.cc(145)] open /sys/devices/system/cpu/cpu0/cpufreq/scaling_cur_freq: No such file or directory (2)
[1219/155817.775587:ERROR:file_io_posix.cc(145)] open /sys/devices/system/cpu/cpu0/cpufreq/scaling_max_freq: No such file or directory (2)
```

I am not quite sure how to address this.

**Comment by gururajh** [comment]
I am able to run npx playwright test on the example tests generated in 24.04LTS - but Tests are failing for Webkit. 

any solutions ?

**Comment by FleetAdmiralJakob** [comment]
`pnpm exec playwright install --with-deps` on Ubuntu 24.04 errors out with

```
Package libasound2 is a virtual package provided by:
  liboss4-salsa-asound2 4.2-build2020-1ubuntu3
  libasound2t64 1.2.11-1build2 (= 1.2.11-1build2)

Package libicu70 is not available, but is referred to by another package.
This may mean that the package is missing, has been obsoleted, or
is only available from another source

E: Package 'libasound2' has no installation candidate
E: Package 'libicu70' has no installation candidate
E: Unable to locate package libffi7
E: Unable to locate package libx264-1[63](https://github.com/The-Creative-Programming-Group/Weather-App/actions/runs/13151049931/job/36698419633#step:4:64)
Failed to install browsers
Error: Installation process exited with code: 100
```

**Comment by Romarionijim** [comment]
I started to have this issue in the last 3 days
I am using ubunut-latest in github actions and I have the playwright version of 1.44.1
I upgraded to the latest playwright version of 1.50.1 and I still recieve the same error in CI with ubuntu latest
dowmngrading to ubuntu 22.04 solved it but I want to use the ubuntu latest since all of our pipelines use it

**Comment by dominicarrojado** [comment]
Faced the same issue today. So in my Github action template I updated the following from:

```
    runs-on: ubuntu-latest
```

to:

```
    runs-on: ubuntu-22.04
````

And that fixed it for now

**Comment by FleetAdmiralJakob** [comment]
But I thought Ubuntu 24 was supported now. It doesn't really fix the issue since it is about the Ubuntu 24 support, and it's just going back to Ubuntu 22.

**Comment by mossymaker** [comment]
Ubuntu 24 works with playwright 1.50.1. Upgrading to the latest release is necessary.

**Comment by brainfoolong** [comment]
For me, even with `mcr.microsoft.com/playwright:v1.51.0-noble` this still produces errors and i cannot use it.

![Image](https://github.com/user-attachments/assets/a6097a2c-6899-4639-a8b7-d9ec9fda1860)

**Comment by mirao** [comment]
@HarrisonUnifyAI 
I suggest to create a new issue for it. Maybe there is an issue with some package configuration in Ubuntu 24.04. But in general the Playwright support for Ubuntu 24.04 was already done and works for some people (even for me).

**Comment by stavros-tomas** [comment]
Hi all, 

Can we please re-open this issue, as it seems still a lot of people are having an issue with this.

I am on ubuntu 24.04.02.
With `npm init playwright@latest` (this translates to 1.51.1), I get the error of `missing dependencies to run browsers` which `npx playwright install --with-deps ` does not solve. 
I also tried manually to install `sudo apt-get update && sudo apt-get install -y  libglib2.0-0 ....` with no luck.

It happens to have another laptop with `ubuntu 22.04` no issue there whatsoever.

**Comment by claurennt** [comment]
> Hi all,
> 
> Can we please re-open this issue, as it seems still a lot of people are having an issue with this.
> 
> I am on ubuntu 24.04.02. With `npm init playwright@latest` (this translates to 1.51.1), I get the error of `missing dependencies to run browsers` which `npx playwright install --with-deps ` does not solve. I also tried manually to install `sudo apt-get update && sudo apt-get install -y libglib2.0-0 ....` with no luck.
> 
> It happens to have another laptop with `ubuntu 22.04` no issue there whatsoever.

Hi there,

upgrading @playwright/test to ^1.52.0 worked for me.

**Comment by adamthewilliam** [comment]
> > Hi all,
> > Can we please re-open this issue, as it seems still a lot of people are having an issue with this.
> > I am on ubuntu 24.04.02. With `npm init playwright@latest` (this translates to 1.51.1), I get the error of `missing dependencies to run browsers` which `npx playwright install --with-deps ` does not solve. I also tried manually to install `sudo apt-get update && sudo apt-get install -y libglib2.0-0 ....` with no luck.
> > It happens to have another laptop with `ubuntu 22.04` no issue there whatsoever.
> 
> Hi there,
> 
> upgrading @playwright/test to ^1.52.0 worked for me.

Thanks, this fixed the issue for me.

**Comment by PierreOlivier-Kompozite** [comment]
For those using a Python container, switching from python:3.12-slim to python:3.12-bullseye fixed the issue for me.

**Comment by tassoman** [comment]
I'm now stuck in Debian testing (forky), and having playwright v.1.54.0. `libasound2` is a meta package.
It should use: `oss4-salsa-asound2` or `libasound2t64`

**Comment by ilvez** [comment]
Same issue on Debian unstable, playwright 1.55.0
